fml/platform/darwin/weak_nsobject.h

```diff
@@ -217,8 +222,9 @@ class WeakNSObject<id> : public WeakNSProtocol<id> {
  private:
   friend class WeakNSObjectFactory<id>;
 
-  explicit WeakNSObject(RefPtr<fml::WeakContainer> container, debug::DebugThreadChecker checker)
-      : WeakNSProtocol<id>(container, checker) {}
+  explicit WeakNSObject(RefPtr<fml::WeakContainer> container,
+                        const debug::DebugThreadChecker& checker)
+      : WeakNSProtocol<id>(std::move(container), checker) {}
```

Comment:
Actually, I didnt catch this first time going through. We should not move here as the factory might spawn multiple weak pointers. Same above for the weak_ptr.

Reply:
Agreed, changing to reference.